iree/base/dynamic_library_win32.cc

```diff
@@ -46,15 +70,80 @@ class DynamicLibraryWin : public DynamicLibrary {
            << "Unable to open dynamic library, not found on search paths";
   }
 
+#if defined(IREE_HAVE_DYNAMIC_LIBRARY_PDB_SUPPORT)
+  void AttachDebugDatabase(const char* database_file_name) override {
+    IREE_TRACE_SCOPE();
+
+    // Derive the base module name (path stem) for the loaded module.
+    // For example, the name of 'C:\Dev\foo.dll' would be 'foo'.
+    // This name is used by dbghelp for listing loaded modules and we want to
+    // ensure we match the name of the PDB module with the library module.
+    std::string module_name = file_name_;
+    size_t last_slash = module_name.find_last_of('\\');
+    if (last_slash != std::string::npos) {
+      module_name = module_name.substr(last_slash + 1);
+    }
+    size_t dot = module_name.find_last_of('.');
+    if (dot != std::string::npos) {
+      module_name = module_name.substr(0, dot);
+    }
```

Comment:
We have a helper for this already: https://github.com/google/iree/blob/f9c11f20fc97154a7079a3809025241f5f467183/iree/base/file_path.h#L38-L39 (implementation in https://github.com/google/iree/blob/main/iree/base/file_path.cc). Want to use that? It would require some fiddling with `/` vs `\\`.

The code is simple, so forking here is fine with me. Keeping path manipulation code in one place makes it easier [to test](https://github.com/google/iree/blob/main/iree/base/file_path_test.cc) though.

Reply:
Yeah it's a \ / issue - having a decent ConvertPathToNative() that would switch based on platform would make this better, but I wanted to see the different ways we'd use it first :)